### ISSUE TITLE
Dependency roles are repeated on different plays

### ISSUE THREAD

**Issue by yajo** [ansible/ansible#44395]
##### SUMMARY

If a server has 2 different roles that both depend on the same 3rd role, the 3rd is executed twice when applying the 2nd role.

According to https://docs.ansible.com/ansible/2.6/user_guide/playbooks_reuse_roles.html#role-dependencies, dependency roles shouldn't be repeated.

##### ISSUE TYPE
 - ~Bug  Report~
- Docs Issue

##### COMPONENT NAME

roles

##### ANSIBLE VERSION

```
ansible 2.6.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/yajo/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, May 16 2018, 17:50:09) [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]
```

##### CONFIGURATION

Nothing special.

##### OS / ENVIRONMENT

Controlling from Fedora 28 an Ubuntu 18.04 server.

##### STEPS TO REPRODUCE

See this [playbook.zip](https://github.com/ansible/ansible/files/2302340/playbook.zip)

##### EXPECTED RESULTS

The `base` role should execute only once per server.

##### ACTUAL RESULTS

The `base` role is executed twice in `server-5.example.net`.

```
➤ ansible-playbook site.yaml -i production 

PLAY [dbservers] ***************************************************************

TASK [Gathering Facts] *********************************************************
ok: [server-2.example.net]
ok: [server-5.example.net]

TASK [base : debug] ************************************************************
ok: [server-2.example.net] => {
    "msg": "I'm a basic server"
}
ok: [server-5.example.net] => {
    "msg": "I'm a basic server"
}

TASK [dbserver : debug] ********************************************************
ok: [server-2.example.net] => {
    "msg": "I'm a db server"
}
ok: [server-5.example.net] => {
    "msg": "I'm a db server"
}

PLAY [webservers] **************************************************************

TASK [Gathering Facts] *********************************************************
ok: [server-5.example.net]

TASK [base : debug] ************************************************************
ok: [server-5.example.net] => {
    "msg": "I'm a basic server"
}

TASK [webserver : debug] *******************************************************
ok: [server-5.example.net] => {
    "msg": "I'm a web server"
}

PLAY RECAP *********************************************************************
server-2.example.net   : ok=3    changed=0    unreachable=0    failed=0   
server-5.example.net   : ok=6    changed=0    unreachable=0    failed=0   
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/playbook/role](https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/role)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by bcoca** [comment]
Possible Misunderstanding
=========================

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We believe the ticket you have filed is being somewhat misunderstood, as one thing works a little differently than stated.

Role de-duplication is PER play, not across plays, so this is expected behaviour.

In the future, this might be a topic more well suited for the user list, which you can also post here if you'd like some more help with the above.
   
   * https://groups.google.com/forum/#!forum/ansible-project

Thank you once again for this and your interest in Ansible!

**Comment by bcoca** [comment]
reopening as docs issue as this is not clear in some sections.

**Comment by nebulon42** [comment]
IMO the documentation does not reflect that fact. I have just stumbled upon it as well. I only learned about this behaviour in this [comment](https://github.com/ansible/ansible/issues/5971#issuecomment-34881242) and then found this issue as well. For me that was a bit unexpected (maybe wrongly so) and so I think it should be clearly written somewhere in the documentation that dependency tracking is PER PLAY only.
